.github/workflows/ci.yml

```diff
@@ -21,24 +21,33 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: recursive
+
       - env:
-          stepName: Build Unit Tests
+          stepName: Install Dependencies
         name: ${{ env.stepName }}
         run: |
           # ${{ env.stepName }}
-          echo "::group::Install Dependencies"
-          git submodule update --checkout
-          cd tools/CMock
-          git submodule update --init vendor/unity
-          cd ../..
+          echo -e "::group::${{ env.bashInfo }} ${{ env.stepName }} ${{ env.bashEnd }}"
+
+          git submodule update --checkout --init --recursive
+          sudo apt-get update -y
```

Comment:
This is a functional change I've added to the PR. Supposed to perform a `sudo apt update` before installing dependencies, for more info refer [here](https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners/customizing-github-hosted-runners#installing-software-on-ubuntu-runners).